### PR TITLE
Fix support for Ubuntu 15.04 and add Debian Jessie support.

### PR DESCRIPTION
**Pull request by VictorLowther**
While we are at it, fix some bitrot along the way.

### CODE REVIEW

File: chef/cookbooks/provisioner/templates/default/net-post-install.sh.erb [hunk 1]
```diff
@@ -109,9 +118,13 @@ echo "PermitRootLogin without-password" >> /etc/ssh/sshd_config
...
 echo "AcceptEnv http_proxy https_proxy no_proxy" >> /etc/ssh/sshd_config
 
-<% if @install_name =~ /^debian-/ -%>
-update-rc.d crowbar_join.sh defaults
-<% end -%>
+if [[ -x /usr/sbin/update-rc.d ]]; then
+    update-rc.d crowbar_join defaults
+else
+    ln -s /etc/init.d/crowbar_join /etc/rc3.d/S80crowbar_join
+    ln -s /etc/init.d/crowbar_join.sh /etc/rc5.d/S80crowbar_join
```
Comment (galthaus) [inline chef/cookbooks/provisioner/templates/default/net-post-install.sh.erb]: I think the two with .sh need to be cut down.

Comment (VictorLowther) [inline chef/cookbooks/provisioner/templates/default/net-post-install.sh.erb]: In fact, update-rc.d is available on all the debian-ish distros we profess to support, so we should be able to just drop the other code, which probably dates back to trying to support 10.10

Comment (galthaus) [inline chef/cookbooks/provisioner/templates/default/net-post-install.sh.erb]: Less code good.

Comment (VictorLowther) [inline chef/cookbooks/provisioner/templates/default/net-post-install.sh.erb]: I will test with the other branch dropped.

**Pull request merged**